src/providers/TxHistoryProvider.tsx

```diff
@@ -0,0 +1,145 @@
+import { TransactionResponse } from '@ethersproject/providers'
+import { TransactionLog, TxStatus } from 'models/transaction'
+import { ReactNode, useCallback, useEffect, useState } from 'react'
+import { clearInterval, setInterval } from 'timers'
+
+import { readProvider } from '../constants/readProvider'
+import { TxHistoryContext } from '../contexts/txHistoryContext'
+
+const nowSeconds = () => Math.round(new Date().valueOf() / 1000)
+
+const KEY_TRANSACTIONS = 'transactions'
+
+const SHORT_TERM_POLL_INTERVAL_MILLIS = 3 * 1000 // 3 sec
+const LONG_TERM_POLL_INTERVAL_MILLIS = 12 * 1000 // 12 sec
+
+// Arbitrary time to give folks a sense of tx history
+const TX_HISTORY_TIME_SECS = 60 * 60 // 1 hr
```

Comment:
refreshing the app will discard any txs older than 1 hour from localStorage

Reply:
totally arbitrary. idea is to give folks some sense of their history if they've been in a sesh on the jb app. but you prolly won't care about old transactions if it's the next day or whatev

Reply:
Could definitely see value in one day adding a "See all transactions" and we just build a list of them based on the connected wallet, but definitely out of scope